src/Microsoft.Graph.Core/Requests/GraphClientFactory.cs

```diff
@@ -0,0 +1,259 @@
+        {
+            if (innerHandler == null)
+            {
+                innerHandler = new HttpClientHandler();
+            }
+
+            if (handlers == null)
+            {
+                return innerHandler;
+            }
+
+            HttpMessageHandler pipeline = innerHandler;
+            IEnumerable<DelegatingHandler> reversedHandlers = handlers.Reverse();
+            foreach (DelegatingHandler handler in reversedHandlers)
+            {
+                if (handler == null)
+                {
+                    throw new ServiceException(
+                    new Error
+                    {
+                        Code = ErrorConstants.Codes.DelegatingHandlerArray,
+                        Message = ErrorConstants.Messages.DelegatingHandlerArrayContainsNullItem,
+                    });
+                }
+
+                if (handler.InnerHandler != null)
+                {
+                    throw new ServiceException(
+                   new Error
+                   {
+                       Code = ErrorConstants.Codes.DelegatingHandlerArrayInnerHandler,
+                       Message = ErrorConstants.Messages.DelegatingHandlerArrayHasNullInnerHandler,
+                   });
+                }
+
+                handler.InnerHandler = pipeline;
+                pipeline = handler;
+            }
+
+            return pipeline;
+        }
+
+
+        /// <summary>
+        /// Configure an instance of an <see cref="HttpClient"/>
+        /// </summary>
+        /// <param name="client">The <see cref="HttpClient"/> client instance need to be configured.</param>
+        /// <param name="timeout">A <see cref="TimeSpan"/> value for the HTTP client timeout property.</param>
+        /// <param name="baseAddress">A <see cref="Uri"/> value to set the HTTP client BaseAddress property.</param>
+        /// <param name="cacheControlHeaderValue">A <see cref="CacheControlHeaderValue"/> value to set HTTP client DefaultRequestHeaders property.</param>
+        /// <returns></returns>
+        private static HttpClient Configure(HttpClient client, TimeSpan timeout, Uri baseAddress, CacheControlHeaderValue cacheControlHeaderValue)
+        {
+            try
+            {
+                client.Timeout = timeout;
+            }
+            catch (InvalidOperationException exception)
+            {
+                throw new ServiceException(
+                    new Error
+                    {
+                        Code = ErrorConstants.Codes.NotAllowed,
+                        Message = ErrorConstants.Messages.OverallTimeoutCannotBeSet,
+                    },
+                    exception);
+            }
+
+            client.BaseAddress = baseAddress == null ? _baseAddress : baseAddress;
+            client.DefaultRequestHeaders.CacheControl = cacheControlHeaderValue ?? new CacheControlHeaderValue { NoCache = true, NoStore = true };
```

Comment:
@MIchaelMainer Do you have history on why this is the default value for CacheControlHeader?

Reply:
That predates me. My guess is that the resources provided by Graph are expected to change so much that we don't ever want a cache copy.